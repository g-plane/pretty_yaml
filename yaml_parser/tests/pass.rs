use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
