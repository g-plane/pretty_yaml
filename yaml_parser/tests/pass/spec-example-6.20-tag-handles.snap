---
source: yaml_parser/tests/pass.rs
---
ROOT@0..52
  DOCUMENT@0..51
    DIRECTIVE@0..34
      PERCENT@0..1 "%"
      TAG_DIRECTIVE@1..34
        DIRECTIVE_NAME@1..4 "TAG"
        WHITESPACE@4..5 " "
        TAG_HANDLE@5..8
          TAG_HANDLE_NAMED@5..8 "!e!"
        WHITESPACE@8..9 " "
        TAG_PREFIX@9..34 "tag:example.com,2000: ..."
    WHITESPACE@34..35 "\n"
    DIRECTIVES_END@35..38 "---"
    WHITESPACE@38..39 "\n"
    FLOW@39..51
      PROPERTIES@39..45
        TAG_PROPERTY@39..45
          SHORTHAND_TAG@39..45
            TAG_HANDLE@39..42
              TAG_HANDLE_NAMED@39..42 "!e!"
            TAG_CHAR@42..45 "foo"
      WHITESPACE@45..46 " "
      DOUBLE_QUOTED_SCALAR@46..51 "\"bar\""
  WHITESPACE@51..52 "\n"
